packages/@ourworldindata/core-table/src/CoreTableColumns.ts

```diff
@@ -266,7 +267,22 @@ export abstract class AbstractCoreColumn<JS_TYPE extends PrimitiveType> {
     }
 
     @imemo get displayName(): string {
-        return this.display?.name ?? this.name ?? ""
+        return (
+            this.display?.name ??
+            this.def.presentation?.titlePublic ?? // this is a bit of an unusual fallback - if display.name is not given, titlePublic is the next best thing before name
+            this.name ??
+            ""
+        )
+    }
+
+    @imemo get titlePublicOrDisplayName(): IndicatorTitleWithFragments {
+        return this.def.presentation?.titlePublic
+            ? {
+                  title: this.def.presentation?.titlePublic,
+                  attributionShort: this.def.presentation?.attributionShort,
+                  titleVariant: this.def.presentation?.titleVariant,
+              }
+            : { title: this.display?.name ?? this.name ?? "" }
     }
 
     @imemo get nonEmptyDisplayName(): string {
```

Comment:
After your changes, this is the only place where `nonEmptyDisplayName` is used now: https://github.com/owid/owid-grapher/blob/cc7ff708a2121cab3ef7d36afc621edf67445942/packages/%40ourworldindata/grapher/src/core/Grapher.tsx#L1467

It makes sense to replace this with `titlePublicOrDisplayName` and then delete `nonEmptyDisplayName`